ghga_connector/core/batch_processing.py

```diff
@@ -98,25 +103,63 @@ def update_staged_files(self):
             )
             sleep(self.retry_after)
 
+    def _check_existing_files(self, *, output_dir: Path):
+        """Check if files would be overwritten and give user a choice if true"""
+        existing_files = {}
+
+        # check local files with and without extension
+        for file_id, file_extension in self.file_ids_with_extension.items():
+            if file_extension:
+                file = output_dir / f"{file_id}.{file_extension}.c4gh"
+            else:
+                file = output_dir / f"{file_id}.c4gh"
+
+            if file.exists():
+                existing_files[file_id] = str(file)
+
+        if existing_files:
+            self._handle_existing_files(files=list(existing_files.values()))
+            # remove existing file ids from those that should be staged
+            for file_id in existing_files:
+                del self.file_ids_with_extension[file_id]
+
+    def _handle_existing_files(self, *, files: list[str]):
+        """Prompt user interaction for already existing files"""
+
+        existing_files = "\n -".join(files)
+        message = f"The following files already exist locally:\n -{existing_files}"
+        self.message_display.display(message)
+
+        already_existing_files = (
+            "Some files already exist locally and will be skipped."
+            + "To redownload files, remove the already present file."
+            + "Do you want to proceed with the remaining files?\n[Yes][No]\n"
+        )
+        response = _get_input(message=already_existing_files)
```

Comment:
The method of getting the input is part of the API and might vary depending on the details of the UI. Should not be part of the core. Added this to the refactoring list.

Reply:
Currently still resides in core as there's some stuff in the CLI/core split that needs to be refactored.
Created some abstractions with ABCs to make this injectable & movable.